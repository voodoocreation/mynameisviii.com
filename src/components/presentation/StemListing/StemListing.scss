@import "../../../scss/variables", "../../../scss/mixins";

.StemListing {
  flex-basis: 100%;
  padding: $spacing-half;

  h2 {
    margin: 0 0 $spacing-half;
  }

  a {
    @include panel-1();

    display: flex;
    flex-direction: row-reverse;
    position: relative;
  }

  &--details {
    flex-grow: 1;
    padding: $spacing;
  }

  &--licence {
    position: absolute;
    bottom: $spacing;
    right: $spacing;
    font-size: 14px;
  }

  &--thumbnail {
    display: none;
  }

  &--image {
    @include square-image();
  }

  &--downloadIcon {
    @include panel-1();

    display: flex;
    position: absolute;
    align-items: center;
    justify-content: center;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    font-size: 64px;
    transform: translate3d(100%, 0, 0);
    transition: transform 200ms ease-out;
  }

  a:hover &--downloadIcon {
    transform: translate3d(0, 0, 0);
  }

  @include when-above(340px) {
    &--thumbnail {
      display: block;
      position: relative;
      flex-grow: 0;
      flex-basis: 36%;
      overflow: hidden;
    }
  }

  @include when-above(480px) {
    &--thumbnail {
      flex-basis: 25%;
    }
  }

  @include when-above(622px) {
    flex-basis: 50%;

    &-thumbnail {
      flex-basis: 38%;
    }
  }

  @include when-above($screen-sm-min) {
    flex-basis: 100%;

    &-thumbnail {
      flex-basis: 20%;
    }
  }

  @include when-above(850px) {
    flex-basis: 50%;

    &-thumbnail {
      flex-basis: 38%;
    }
  }

  @include when-above(1048px) {
    &-thumbnail {
      flex-basis: 33.333333%;
    }
  }

  @include when-above(1330px) {
    flex-basis: 33.333333%;

    &-thumbnail {
      flex-basis: 38%;
    }
  }

  @include when-above(1784px) {
    &-thumbnail {
      flex-basis: 25%;
    }
  }
}
