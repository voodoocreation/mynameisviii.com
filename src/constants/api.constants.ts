export enum BOOLEAN {
  TRUE = "y",
  FALSE = "n",
}
