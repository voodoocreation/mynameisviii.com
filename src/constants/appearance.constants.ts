export enum STATUS {
  CANCELLED = "EventCancelled",
  POSTPONED = "EventPostponed",
  RESCHEDULED = "EventRescheduled",
  SCHEDULED = "EventScheduled"
}

export enum TYPE {
  ART = "VisualArtsEvent",
  EXHIBITION = "ExhibitionEvent",
  FESTIVAL = "Festival",
  MUSIC = "MusicEvent",
  SCREENING = "ScreeningEvent",
  SOCIAL = "SocialEvent"
}
