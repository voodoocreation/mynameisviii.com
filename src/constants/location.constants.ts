export enum TYPE {
  CITY = "City",
  EVENT_VENUE = "EventVenue",
  MOVIE_THEATER = "MovieTheater",
  MUSIC_VENUE = "MusicVenue",
  PARK = "Park",
  PERFORMING_ARTS_THEATER = "PerformingArtsTheater",
  STADIUM = "StadiumOrArena",
}
