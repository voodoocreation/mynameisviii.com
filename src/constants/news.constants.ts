export enum TYPE {
  APPEARANCE = "appearance",
  NEWS = "news",
  RELEASE = "release"
}
