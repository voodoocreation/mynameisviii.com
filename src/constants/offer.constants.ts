export enum AVAILABILITY {
  IN_STOCK = "InStock",
  IN_STORE = "InStoreOnly",
  LIMITED = "LimitedAvailability",
  OUT_OF_STOCK = "OutOfStock",
  ONLINE = "OnlineOnly",
  PRE_ORDER = "PreOrder",
  PRE_SALE = "PreSale",
  SOLD_OUT = "SoldOut",
}
