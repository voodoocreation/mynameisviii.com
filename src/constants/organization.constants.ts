export enum TYPE {
  CORPORATION = "Corporation",
  CHARITY = "FundingScheme",
  LOCAL_BUSINESS = "LocalBusiness",
  MEDIA = "NewsMediaOrganization",
  PERFORMER = "PerformingGroup",
  PROJECT = "Project",
}
