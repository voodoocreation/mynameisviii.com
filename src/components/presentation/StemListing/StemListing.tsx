import cn from "classnames";
import * as React from "react";
import {
  FaCreativeCommons,
  FaCreativeCommonsBy,
  FaCreativeCommonsNc,
  FaCreativeCommonsSa,
} from "react-icons/fa";
import { MdCloudDownload } from "react-icons/md";
import { injectIntl, WrappedComponentProps } from "react-intl";

import { IStem } from "../../../models/root.models";
import Image from "../Image/Image";
import Link from "../Link/Link";
import Meta from "../Meta/Meta";

import "./StemListing.scss";

interface IProps extends IStem, WrappedComponentProps {
  onLoad: () => void;
}

interface IState {
  isRendered: boolean;
}

class StemListing extends React.Component<IProps, IState> {
  public readonly state: IState = {
    isRendered: false,
  };

  public render() {
    const { intl, onLoad, ...stem } = this.props;
    const { formatMessage } = intl;
    const { isRendered } = this.state;

    return (
      <article className={cn("StemListing", { isRendered })}>
        <Link
          className="StemListing--link"
          href={stem.url}
          title={formatMessage(
            { id: "DOWNLOAD_STEMS_FOR" },
            { title: stem.title }
          )}
          isExternal
        >
          <div className="StemListing--details">
            <header className="StemListing--header">
              <h2>
                <span>{stem.title}</span>
              </h2>
            </header>

            <section className="StemListing--meta">
              <Meta
                className="StemListing--audioFormat"
                labelIntlId="AUDIO_FORMAT"
              >
                {stem.audioFormat}
              </Meta>

              <Meta
                className="StemListing--packageFormat"
                labelIntlId="PACKAGE_FORMAT"
              >
                {stem.packageFormat}, {stem.size}
              </Meta>

              <Meta
                className="StemListing--licence"
                icon={
                  <>
                    <FaCreativeCommons />
                    <FaCreativeCommonsBy />
                    <FaCreativeCommonsNc />
                    <FaCreativeCommonsSa />
                  </>
                }
                labelIntlId="LICENCED_UNDER_CC_REMIX"
                title={formatMessage({ id: "LICENCED_UNDER_CC_REMIX" })}
              />
            </section>
          </div>

          <div className="StemListing--thumbnail">
            <Image
              alt={stem.title}
              className="StemListing--image"
              src={stem.imageUrl}
              onLoad={this.onLoad}
            />

            <div className="StemListing--downloadIcon">
              <MdCloudDownload />
            </div>
          </div>
        </Link>
      </article>
    );
  }

  private onLoad = () => {
    this.setState({
      isRendered: true,
    });

    if (this.props.onLoad) {
      this.props.onLoad();
    }
  };
}

export default injectIntl(StemListing);
