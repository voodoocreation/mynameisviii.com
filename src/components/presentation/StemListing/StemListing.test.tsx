import { PACKAGE_FORMAT } from "../../../constants/stem.constants";
import { stem } from "../../../models/root.models";
import WrapperWithIntl from "../../../utilities/WrapperWithIntl";
import StemListing from "./StemListing";

const component = new WrapperWithIntl(StemListing).withDefaultProps(
  stem({
    audioFormat: "Audio format",
    createdAt: "2017-10-10T18:00:00",
    imageUrl: "Image URL",
    packageFormat: PACKAGE_FORMAT.RAR,
    size: "Size",
    slug: "stem-1",
    title: "Title",
    url: "URL"
  })
);

describe("[presentation] <StemListing />", () => {
  const wrapper = component
    .withProps({
      onLoad: jest.fn()
    })
    .mount();

  it("doesn't add the isRendered class initially", () => {
    expect(wrapper.find(".isRendered")).toHaveLength(0);
  });

  it("loads the image", () => {
    wrapper.find("Image.StemListing--image").simulate("load");
  });

  it("adds the isRendered class", () => {
    expect(wrapper.find(".isRendered")).toHaveLength(1);
  });

  it("calls the onLoad prop", () => {
    expect(component.props.onLoad).toHaveBeenCalledTimes(1);
  });

  it("sets the onLoad prop to be undefined", () => {
    wrapper.setProps({ onLoad: undefined });
  });

  it("loads the image", () => {
    wrapper.find("Image.StemListing--image").simulate("load");
  });

  it("matches snapshot", () => {
    expect(wrapper.render()).toMatchSnapshot();
  });
});
