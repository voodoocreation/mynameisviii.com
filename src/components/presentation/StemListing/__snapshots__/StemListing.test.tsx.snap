// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[presentation] <StemListing /> matches snapshot 1`] = `
<article
  class="StemListing isRendered"
>
  <a
    class="StemListing--link"
    href="URL"
    rel="noopener"
    target="_blank"
    title="Download stems for {title}"
  >
    <div
      class="StemListing--details"
    >
      <header
        class="StemListing--header"
      >
        <h2>
          <span>
            Title
          </span>
        </h2>
      </header>
      <section
        class="StemListing--meta"
      >
        <div
          class="Meta StemListing--audioFormat"
        >
           
          <span
            class="Meta--label"
          >
            Audio format:
          </span>
           
          <span
            class="Meta--value"
          >
            Audio format
          </span>
        </div>
        <div
          class="Meta StemListing--packageFormat"
        >
           
          <span
            class="Meta--label"
          >
            Package format:
          </span>
           
          <span
            class="Meta--value"
          >
            RAR, Size
          </span>
        </div>
        <div
          class="Meta StemListing--licence"
          title="Licenced under the Creative Commons Attribution-NonCommercial-ShareAlike licence (CC BY-NC-SA)"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M245.83 214.87l-33.22 17.28c-9.43-19.58-25.24-19.93-27.46-19.93-22.13 0-33.22 14.61-33.22 43.84 0 23.57 9.21 43.84 33.22 43.84 14.47 0 24.65-7.09 30.57-21.26l30.55 15.5c-6.17 11.51-25.69 38.98-65.1 38.98-22.6 0-73.96-10.32-73.96-77.05 0-58.69 43-77.06 72.63-77.06 30.72-.01 52.7 11.95 65.99 35.86zm143.05 0l-32.78 17.28c-9.5-19.77-25.72-19.93-27.9-19.93-22.14 0-33.22 14.61-33.22 43.84 0 23.55 9.23 43.84 33.22 43.84 14.45 0 24.65-7.09 30.54-21.26l31 15.5c-2.1 3.75-21.39 38.98-65.09 38.98-22.69 0-73.96-9.87-73.96-77.05 0-58.67 42.97-77.06 72.63-77.06 30.71-.01 52.58 11.95 65.56 35.86zM247.56 8.05C104.74 8.05 0 123.11 0 256.05c0 138.49 113.6 248 247.56 248 129.93 0 248.44-100.87 248.44-248 0-137.87-106.62-248-248.44-248zm.87 450.81c-112.54 0-203.7-93.04-203.7-202.81 0-105.42 85.43-203.27 203.72-203.27 112.53 0 202.82 89.46 202.82 203.26-.01 121.69-99.68 202.82-202.84 202.82z"
            />
          </svg>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M314.9 194.4v101.4h-28.3v120.5h-77.1V295.9h-28.3V194.4c0-4.4 1.6-8.2 4.6-11.3 3.1-3.1 6.9-4.7 11.3-4.7H299c4.1 0 7.8 1.6 11.1 4.7 3.1 3.2 4.8 6.9 4.8 11.3zm-101.5-63.7c0-23.3 11.5-35 34.5-35s34.5 11.7 34.5 35c0 23-11.5 34.5-34.5 34.5s-34.5-11.5-34.5-34.5zM247.6 8C389.4 8 496 118.1 496 256c0 147.1-118.5 248-248.4 248C113.6 504 0 394.5 0 256 0 123.1 104.7 8 247.6 8zm.8 44.7C130.2 52.7 44.7 150.6 44.7 256c0 109.8 91.2 202.8 203.7 202.8 103.2 0 202.8-81.1 202.8-202.8.1-113.8-90.2-203.3-202.8-203.3z"
            />
          </svg>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M247.6 8C387.4 8 496 115.9 496 256c0 147.2-118.5 248-248.4 248C113.1 504 0 393.2 0 256 0 123.1 104.7 8 247.6 8zM55.8 189.1c-7.4 20.4-11.1 42.7-11.1 66.9 0 110.9 92.1 202.4 203.7 202.4 122.4 0 177.2-101.8 178.5-104.1l-93.4-41.6c-7.7 37.1-41.2 53-68.2 55.4v38.1h-28.8V368c-27.5-.3-52.6-10.2-75.3-29.7l34.1-34.5c31.7 29.4 86.4 31.8 86.4-2.2 0-6.2-2.2-11.2-6.6-15.1-14.2-6-1.8-.1-219.3-97.4zM248.4 52.3c-38.4 0-112.4 8.7-170.5 93l94.8 42.5c10-31.3 40.4-42.9 63.8-44.3v-38.1h28.8v38.1c22.7 1.2 43.4 8.9 62 23L295 199.7c-42.7-29.9-83.5-8-70 11.1 53.4 24.1 43.8 19.8 93 41.6l127.1 56.7c4.1-17.4 6.2-35.1 6.2-53.1 0-57-19.8-105-59.3-143.9-39.3-39.9-87.2-59.8-143.6-59.8z"
            />
          </svg>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M247.6 8C389.4 8 496 118.1 496 256c0 147.1-118.5 248-248.4 248C113.6 504 0 394.5 0 256 0 123.1 104.7 8 247.6 8zm.8 44.7C130.2 52.7 44.7 150.6 44.7 256c0 109.8 91.2 202.8 203.7 202.8 103.2 0 202.8-81.1 202.8-202.8.1-113.8-90.2-203.3-202.8-203.3zM137.7 221c13-83.9 80.5-95.7 108.9-95.7 99.8 0 127.5 82.5 127.5 134.2 0 63.6-41 132.9-128.9 132.9-38.9 0-99.1-20-109.4-97h62.5c1.5 30.1 19.6 45.2 54.5 45.2 23.3 0 58-18.2 58-82.8 0-82.5-49.1-80.6-56.7-80.6-33.1 0-51.7 14.6-55.8 43.8h18.2l-49.2 49.2-49-49.2h19.4z"
            />
          </svg>
           
          <span
            class="Meta--label"
          >
            Licenced under the Creative Commons Attribution-NonCommercial-ShareAlike licence (CC BY-NC-SA):
          </span>
           
          <span
            class="Meta--value"
          />
        </div>
      </section>
    </div>
    <div
      class="StemListing--thumbnail"
    >
      <figure
        class="Image StemListing--image isLoading"
      >
        <div
          class="Image--wrapper"
        >
          <img
            alt="Title"
            src="Image URL"
          />
          <span
            class="Loader"
          >
            <div
              class="Loader--animation"
            >
              <span />
              <span />
              <span />
            </div>
          </span>
        </div>
      </figure>
      <div
        class="StemListing--downloadIcon"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM17 13l-5 5-5-5h3V9h4v4h3z"
          />
        </svg>
      </div>
    </div>
  </a>
</article>
`;
